presto-spi/src/main/java/com/facebook/presto/spi/function/SqlInvokedFunction.java

```diff
@@ -98,10 +121,19 @@ public SqlInvokedFunction(
                 .map(Parameter::getType)
                 .collect(collectingAndThen(toList(), Collections::unmodifiableList));
 
-        this.signature = new Signature(functionName, SCALAR, typeVariableConstraints, emptyList(), returnType, argumentTypes, false);
+        this.signature = new Signature(functionName, kind, typeVariableConstraints, emptyList(), returnType, argumentTypes, false);
         this.functionId = new SqlFunctionId(functionName, argumentTypes);
         this.functionVersion = requireNonNull(version, "version is null");
         this.functionHandle = version.hasVersion() ? Optional.of(new SqlFunctionHandle(this.functionId, version.toString())) : Optional.empty();
+
+        if (kind == AGGREGATE && (aggregationMetadata == null || !aggregationMetadata.isPresent())) {
```

Comment:
I think I would do a checkArgument here instead.  Something like:

```
checkArgument(kind == AGGREGATE && aggregationMetadata.isPresent() || kind != AGGREGATE && !aggregationMetadata.isPresent(), "aggregationMetadata must be present for aggregation functions and absent for non-aggregation functions.")
```

Reply:
`com.google.common.base.Preconditions.checkArgument` is not accessible in this path & `Utils.checkArgument` does not have a `errorMessage` arg. So, I updated the if-block into 1 if-check with your suggestions. Note that I am still `throw`ing  (`IllegalArgumentException` similar to `checkArgument` impl) since I can't access `checkArgument` as is).

Reply:
thanks.  forgot it wasn't accessible from the spi